File: calculadora.java
// Calculadora feita em java.
import javax.swing.JOptionPane;
public class calculadora {
    public static void main(String[] args){

        int resultado = 0;

        int n1 = Integer.parseInt(JOptionPane.showInputDialog(null, "Digite o primeiro valor"));
        //Método ".charAT(0)" para o JOptionPane receber variáveis do tipo "char".
        char operador = JOptionPane.showInputDialog("Digite o operador:").charAt(0);
        int n2 = Integer.parseInt(JOptionPane.showInputDialog(null, "Digite o segundo valor:"));

        if(operador == '+'){
            resultado = n1 + n2;
        }
        if(operador == '-'){
            resultado = n1 - n2;
        }
        if(operador == '*'){
            resultado = n1 * n2;
        }
        if(operador == '/'){
            resultado = n1 / n2;
        }
        JOptionPane.showMessageDialog(null, resultado);
    }
    
}

File: calculadora.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <math.h>

int main(int argc, char *argv[]){

	float valor1, valor2, resultado; //"Float" serve para variáveis do tipo real.
	char operador[30];

	printf("Digite o primeiro valor:");
	scanf("%f", &valor1); //"%f" serve para valores do tipo float.

	printf("Digite o operador:");
	scanf("%s", &operador); //usar "&" para operadores.

	printf("Digite o segundo valor:");
	scanf("%f", &valor2);

	if(operador == '+'){ //Necessário uso de aspas simples para operadores.
		resultado = valor1 + valor2;
	}
	if(operador == '-'){
		resultado = valor1 - valor2;
	}
	if(operador == '*'){
		resultado = valor1 * valor2;
	}
	if(operador == '/'){
		resultado = valor1 / valor2;
	}
	printf("%f", resultado);


	return 0;
}